docs/architecture/css-architecture.md

```diff
@@ -0,0 +1,121 @@
+
+Additionally, an existing UI framework such as Bootstrap can be used in Spartacus without making it a hardcoded depenendency.
+
+### Style Library Versioning
+
+Spartacus libraries support Semantic Versioning, which means that no breaking changes are allowed during a major version. Uncontrolled style changes in Spartacus would have a bad effect on customers, as unexpected styling could appear without notice. 
+
+To provide a stable style release for a major version of the library, the style library is shipped with the same semantic versioning scheme as the `core` and `component` libraries. Customers can rely on a major version of the style library, so that only new features and bugs can be expected in minor and patch releases. 
+This means that the style rules for a major version cannot get any changes other then new (optional) styles (for example for a new component) or to patch a bug.
+
+In order to continuously improve the styles during a major release, new styles might be isolated in special theme. This theme could be shipped already during the major release. This allows to demonstrate and use right away, without interfering with existing users. 
+
+The table below shows an example of some potential releases. 
+
+| Version | Theme                     | Note                                                                                                         |
+| ------- | ------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| 1.0.0   | sparta                    | standard theme for 1.0.0 release
+| 1.0.1   | sparta                    | *patch* release with a sparta theme fix                                                                      |
+| 1.1.0   | sparta, coconut-beta-1    | new features for sparta and new beta release for new coconut theme                                           |
+| 2.0.0   | cocunut (default), sparta | New major version could promote a new theme a the default. The sparta style could become deprated over time. |
+
+
+
+## CSS Technology
+Spartacus is developed with a combination of CSS techniques:
+-  [SASS]() is used as the pre-processing language, like most of todays UI frameworks do
+-  [CSS variables]() are used for global theming
+-  [CSS post processing]() is intended to be used to polyfil any of required syntax for older browsers.
+
+### View encapsulation
+
+View encapsulation ensures isolation of styles in single DOM. View encapsulation can be used to ensure that the component style rules from one componet do not interfere with other components. View encapsulation is standardised in a web component architecture, profided by the so-called `shadow-dom`. Applications that do not leverage the shadow-dom there are a few alternatives avaiable: 
```

Comment:
View encapsulation can be used to ensure that the component style rules from one `component` do not interfere with other components. View encapsulation is `standardized` in a web component architecture, profided by the so-called shadow-dom.